circle.yml

```diff
@@ -15,6 +15,64 @@ executors:
       xcode: 11.1.0
 
 commands:
+  build_silkworm:
+    steps:
+      - restore_cache:
+          name: "Restore Silkworm cache"
+          key: &silkworm-cache-key silkworm-v1
+      - run:
+          name: "Check Silkworm cache"
+          command: |
+            if [ -f ~/silkworm/consensus ]; then
+              echo 'Cached Silkworm binary available - skip build.'
+            else
+              echo 'export SILKWORM_BUILD=true' >> $BASH_ENV
+            fi
+      - run:
+          name: "Install GMP-dev"
+          command: |
+            [ "$SILKWORM_BUILD" = true ] || exit 0
+            sudo apt-get -q update && sudo apt-get -qy install --no-install-recommends libgmp-dev
+      - run:
+          name: "Checkout Silkworm"
+          working_directory: ~/silkworm/src
+          command: |
+            [ "$SILKWORM_BUILD" = true ] || exit 0
+            git clone --no-checkout --single-branch https://github.com/torquem-ch/silkworm.git .
+            git checkout 614fabfcf19b8e06f0dcbe3e9af2345dcf748f3e
+            git submodule init
+            git submodule deinit tests
+            git submodule update --recursive --depth=1 --progress
+      - run:
+          name: "Configure Silkworm"
+          working_directory: ~/silkworm
+          command: |
+            [ "$SILKWORM_BUILD" = true ] || exit 0
+            cmake -S src -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_RUNTIME_OUTPUT_DIRECTORY=$(pwd)
+      - run:
+          name: "Build Silkworm cmd/consensus"
+          working_directory: ~/silkworm
+          command: |
+            [ "$SILKWORM_BUILD" = true ] || exit 0
+            cmake --build build --parallel --target consensus
+      - save_cache:
+          name: "Save Silkworm cache"
+          key: *silkworm-cache-key
+          paths:
+            - ~/silkworm/consensus
+
+  download_consensus_tests:
```

Comment:
Why not chace this too?

Reply:
Cache means you have to download and unpack it, just from different server.

Reply:
Ah ok, I thought it is some better integration with circleci.